lib/auth/access_request_test.go

```diff
@@ -196,6 +196,43 @@ func TestAccessRequest(t *testing.T) {
 	t.Run("deny", func(t *testing.T) { testAccessRequestDenyRules(t, testPack) })
 }
 
+// waitForAccessRequests is a helper for writing access request tests that need to wait for access request CRUD. the supplied condition is
+// repeatedly called with the contents of the access request cache until it returns true or a reasonably long timeout is exceeded. this is
+// similar to require.Eventually except that it is safe to use normal (test-failing) assertions within the supplied condition closure.
+func waitForAccessRequests(t *testing.T, ctx context.Context, getter services.AccessRequestGetter, condition func([]*types.AccessRequestV3) bool) {
+	t.Helper()
+
+	timeout := time.After(time.Second * 30)
+	for {
+		var reqs []*types.AccessRequestV3
+		var nextKey string
+	Paginate:
+		for {
+			rsp, err := getter.ListAccessRequests(ctx, &proto.ListAccessRequestsRequest{
+				Limit:    1_000,
+				StartKey: nextKey,
+			})
+			require.NoError(t, err, "ListAccessRequests API call should succeed")
+
+			reqs = append(reqs, rsp.AccessRequests...)
+			nextKey = rsp.NextKey
+			if nextKey == "" {
+				break Paginate
+			}
+		}
+
+		if condition(reqs) {
+			return
+		}
+
+		select {
```

Comment:
Do we need a select here? Can we just check `if time.Since(start) > 30 * time.Second`?